### PR TITLE
Using signed ints for the word index. A word index can never be a negative number

### PR DESCRIPTION
**Pull request by Vagelis-Prokopiou**
@JockeTF how about this?
I think it is a more logical solution.
Can you provide your thoughts on it?

### CODE REVIEW

**Comment by JockeTF** [comment]
I'm not sure!

indexes are usually `usize` in Rust. I believe this has to do with the pointer size of machines. For example, if we're on a 32-bit machine we'd at most be able to address 4 GB (`2 ^ 32 = 4 294 967 296`) of memory. We therefore know that an array cannot contain more than `2^32` elements since addressing anything above that would be impossible without resulting to other tricks.

On a 64-bit machine we'd be able to address larger arrays than on 32-bit machines, so `usize` would be 64-bit there. Negative indexes also make no sense unless we use this to mean something else, such as indexing from the end like in Python. Rust usually doesn't do that though, so I don't thing taking signed integers for the index makes sense unless something similar is implemented.

In summary:

- Any values below `usize` won't make sense since we cannot access values before the start of the string.
- Any values above `usize` won't make sense since the machine cannot access such addresses either way.
- Using `i16` or `u16` on a 64-bit address space imposes arbitrary limits, even though real words are shorter.

I think sticking to `usize` makes the most sense since that's what indexes usually are in Rust. Using other types would generally only invite callers to pass values that would break the function. Either way, it would still make sense to add a check ensuring that the caller doesn't pass indexes that are invalid for the string!

**Comment by Vagelis-Prokopiou** [comment]
On seconds thoughts, indeed, I don't like it either.
The mere fact that I am imposing a limit with the signed integer, is unnecessary.
So I will stick with your current fix for the empty string panics.
Thanks  again.
